spec/controllers/api/users_controller_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe API::UsersController, type: :controller do
+  let(:user) { classroom_teacher }
+
+  describe "GET #authenticated_user", :vcr do
+    before do
+      get :authenticated_user, params: {
+        access_token: user.api_token
+      }
+    end
+
+    it "returns success" do
+      expect(response).to have_http_status(:success)
+    end
+
+    context "user serializer returns correct attributes" do
+      it "returns user id" do
+        expect(json["id"]).to eq(user.id)
+      end
+
+      it "returns username" do
+        expect(json["username"]).to eq(user.github_user.login)
```

Comment:
These tests don't need to separated as they are just checking the same json hash. Would be easier to read and write a test that checks the entire json blob. Something like this:
```ruby
describe "success" do
  it "has a status of 200" do
    expect(response).to have_http_status(:success)
  end

  it "returns a structured body" do
    expect(json).to eq(
      "id": user.id,
      "username": user.github_user.login
    )
  end
end
```